// applicationinsights/out/src/metrics/types.js
"use strict";
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.
Object.defineProperty(exports, "__esModule", { value: true });
exports.PerformanceCounterMetricNames = exports.StandardMetricNames = exports.APPLICATION_INSIGHTS_NO_STANDARD_METRICS = void 0;
/**
 * Disable Standard Metrics environment variable name.
 */
exports.APPLICATION_INSIGHTS_NO_STANDARD_METRICS = "APPLICATION_INSIGHTS_NO_STANDARD_METRICS";
var StandardMetricNames;
(function (StandardMetricNames) {
    StandardMetricNames["HTTP_REQUEST_DURATION"] = "azureMonitor.http.requestDuration";
    StandardMetricNames["HTTP_DEPENDENCY_DURATION"] = "azureMonitor.http.dependencyDuration";
    StandardMetricNames["EXCEPTION_COUNT"] = "azureMonitor.exceptionCount";
    StandardMetricNames["TRACE_COUNT"] = "azureMonitor.traceCount";
})(StandardMetricNames = exports.StandardMetricNames || (exports.StandardMetricNames = {}));
var PerformanceCounterMetricNames;
(function (PerformanceCounterMetricNames) {
    PerformanceCounterMetricNames["PRIVATE_BYTES"] = "\\Process(??APP_WIN32_PROC??)\\Private Bytes";
    PerformanceCounterMetricNames["AVAILABLE_BYTES"] = "\\Memory\\Available Bytes";
    PerformanceCounterMetricNames["PROCESSOR_TIME"] = "\\Processor(_Total)\\% Processor Time";
    PerformanceCounterMetricNames["PROCESS_TIME"] = "\\Process(??APP_WIN32_PROC??)\\% Processor Time";
    PerformanceCounterMetricNames["REQUEST_RATE"] = "\\ASP.NET Applications(??APP_W3SVC_PROC??)\\Requests/Sec";
    PerformanceCounterMetricNames["REQUEST_DURATION"] = "\\ASP.NET Applications(??APP_W3SVC_PROC??)\\Request Execution Time";
})(PerformanceCounterMetricNames = exports.PerformanceCounterMetricNames || (exports.PerformanceCounterMetricNames = {}));
